lib/expression/v2.ex

```diff
@@ -73,28 +76,80 @@ defmodule Expression.V2 do
     end
   end
 
-  @spec eval(String.t(), context :: Context.t()) :: [term]
-  def eval(expression, context \\ Context.new()) do
-    parts = compile(expression)
+  @doc """
+  Parse a string with an expression block into AST for the compile step
+  """
+  @spec parse_block(String.t()) ::
+          {:ok, [term]} | {:error, reason :: String.t(), bad_parts :: String.t()}
+  def parse_block(expression_block) do
+    case Parser.expression(expression_block) do
+      {:ok, ast, "", _, _, _} -> {:ok, ast}
+      {:error, _ast, remaining, _, _, _} -> {:error, "Unable to parse remainder", remaining}
+    end
+  end
+
+  @doc """
+  Evaluate a string with an expression block against a context
+  """
+  @spec eval_block(String.t(), context :: Context.t()) :: term
+  def eval_block(expression_block, context \\ Context.new()) do
+    {:ok, ast} = parse_block(expression_block)
```

Comment:
should we handle the **error** case here as well?

Reply:
done in cc5760a